cmd/agent/subcommands/hostname/command.go

```diff
@@ -47,16 +52,55 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 			)
 		},
 	}
+	getHostnameCommand.Flags().BoolVarP(&cliParams.forceLocal, "local", "l", false, "Force computing the hostname in the command line instead of the agent process")
+
 	cliParams.logLevelDefaultOff.Register(getHostnameCommand)
 	return []*cobra.Command{getHostnameCommand}
 }
 
-func getHostname(_ log.Component, _ config.Component, _ *cliParams) error {
-	hname, err := hostname.Get(context.TODO())
-	if err != nil {
-		return fmt.Errorf("Error getting the hostname: %v", err)
+func getHostname(_ log.Component, config config.Component, params *cliParams) error {
+	return getHostnameWithWriter(nil, config, params, os.Stdout)
```

Comment:
Two questions:
- Why pass the `nil log.Component` it has being initialized, so we do not have to pass it to the new function 😄 
- Why creating the  `getHostnameWithWriter` instead of having the code inline in `getHostname`?

Reply:
For the first question it's just because I copy pasted the function, I should definitely remove the logger argument haha
For the second it's just to ease testing, it avoids having to capture stdout to test what was written

Reply:
Fixed !